zebra-state/src/service/finalized_state/zebra_db/block.rs

```diff
@@ -161,6 +175,56 @@ impl ZebraDb {
         }))
     }
 
+    /// Returns the [`Block`] with [`block::Hash`] or [`Height`], if it exists
+    /// in the finalized chain, and its serialized size.
+    #[allow(clippy::unwrap_in_result)]
+    pub fn block_and_size(&self, hash_or_height: HashOrHeight) -> Option<(Arc<Block>, usize)> {
+        let (raw_header, raw_txs) = self.raw_block(hash_or_height)?;
+
+        let header = Arc::<block::Header>::from_bytes(raw_header.raw_bytes());
+        let txs: Vec<_> = raw_txs
+            .iter()
+            .map(|raw_tx| Arc::<Transaction>::from_bytes(raw_tx.raw_bytes()))
+            .collect();
+
+        // Compute the size of the block from the size of header and size of
+        // transactions. This requires summing them all and also adding the
+        // size of the CompactSize-encoded transaction count.
+        // See https://developer.bitcoin.org/reference/block_chain.html#serialized-blocks
+        let tx_count = CompactSizeMessage::try_from(txs.len()).unwrap();
+        let tx_raw = tx_count.zcash_serialize_to_vec().unwrap();
```

Comment:
Will it be better to return `None` instead of panicking ?

Reply:
Blocks deserialized from the database are guaranteed to be serializable without errors. I should have used expect though, I will change it

Reply:
Done in 826c6bf3d